Automattic/wp-calypso
# Launchpad: Use new task definition and tracking for setup_free task

Fix part of #85694

## Proposed Changes

* Transform the free flow tasks to links. 
* Move only the `setup_free`  task definition 
* Enable it only for the free flow
* Improve the user tracking on the new task definitions
* Unify the Task type definition
* [Adopt strangler fig pattern using feature flag to migrate gradually all task definitions ](https://dennylesmana.medium.com/what-is-the-strangler-fig-pattern-1560443b8459#:~:text=The%20Strangler%20Fig%20pattern%20aims,that%20isn't%20used%20yet)
* The remaining flow will be covered by other PRs (way smaller than this one).

---
### Technical Notes
The overall idea is to migrate all of this task definition by flow, testing them gradually.

Since it is a very sensitive product area and with low test coverage, this change is under a feature flag, so  we are going to remove the old code as soon all tasks were migrated


--- 

## Testing Instructions
* Run the /setup/free on you localmachine or calypso.live (the feature flag is only available on this envs)
* Follow the steps until the launchpad is rendered
* Open the console and check if this messages were rendered 
``` 
Using new task definition parser {taskId: 'setup_free', flowId: 'free'}
``` 
* Check if the setup_free (Personalize your site) task is a link
* Check if the process happened with success
* Check if the events now contains the following attributes:
``` 
blog_id: 227838534 
checklist_completed: false
checklist_slug: free
client: browser
context: fullscreen
environment: development
environment_id: development
flow: free
is_completed: true
order: 3
site_count: 256
site_id_label: wpcom
site_intent: free
site_plan_id: 1
task_id: design_edited
vph: 873
vpw: 2560
``` 




*

## Pre-merge Checklist
